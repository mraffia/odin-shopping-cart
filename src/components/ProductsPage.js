import React from 'react';
import '../styles/ProductsPage.css';
import ProductCard from './ProductCard.js';

function ProductsPage({ items, handleAddItem }) {
  return (
    <div className="products-container">
        <div className="products-subcontainer">
          {items.map((item, i) => {
            return (
              <div key={i}>
                <ProductCard item={item} handleAddItem={handleAddItem} />
              </div>
            )
          })}
        </div>
    </div>
  );
}

export default ProductsPage;
