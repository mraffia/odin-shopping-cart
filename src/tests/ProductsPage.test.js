import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProductsPage from '../components/ProductsPage.js';

describe("ProductsPage component", () => {
    it("renders ProductsPage content", () => {
        const items = [
            { name: "Gathering Swarm", price: 300 },
            { name: "Heavy Blow", price: 350 },
            { name: "Lifeblood Heart", price: 250 },
            { name: "Longnail", price: 300 },
            { name: "Quick Focus", price: 800 },
            { name: "Salubra's Blessing", price: 800 },
            { name: "Shaman Stone", price: 220 },
            { name: "Steady Body", price: 120 },
        ];
        const { container } = render(<ProductsPage items={items} />);
    
        expect(container).toMatchSnapshot();
    });
});