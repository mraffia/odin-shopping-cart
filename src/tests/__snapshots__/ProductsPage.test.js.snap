// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPage component renders ProductsPage content 1`] = `
<div>
  <div
    class="products-container"
  >
    <div
      class="products-subcontainer"
    >
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Gathering Swarm"
              class="card-image"
              src="gatheringSwarm.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Gathering Swarm
            </div>
            <div
              class="card-price"
            >
              300
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Heavy Blow"
              class="card-image"
              src="heavyBlow.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Heavy Blow
            </div>
            <div
              class="card-price"
            >
              350
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Lifeblood Heart"
              class="card-image"
              src="lifebloodHeart.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Lifeblood Heart
            </div>
            <div
              class="card-price"
            >
              250
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Longnail"
              class="card-image"
              src="longnail.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Longnail
            </div>
            <div
              class="card-price"
            >
              300
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Quick Focus"
              class="card-image"
              src="quickFocus.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Quick Focus
            </div>
            <div
              class="card-price"
            >
              800
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Salubra's Blessing"
              class="card-image"
              src="salubraGhost.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Salubra's Blessing
            </div>
            <div
              class="card-price"
            >
              800
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Shaman Stone"
              class="card-image"
              src="shamanStone.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Shaman Stone
            </div>
            <div
              class="card-price"
            >
              220
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
      <div>
        <div
          class="card-container"
        >
          <div
            class="card-image-container"
          >
            <img
              alt="Steady Body"
              class="card-image"
              src="steadyBody.png"
            />
          </div>
          <div
            class="card-info-container"
          >
            <div
              class="card-name"
            >
              Steady Body
            </div>
            <div
              class="card-price"
            >
              120
               Geo
            </div>
            <button
              class="card-button"
              type="button"
            >
              Add to cart
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
